src/gc/gcinterface.dac.h

```diff
@@ -0,0 +1,144 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef _GC_INTERFACE_DAC_H_
+#define _GC_INTERFACE_DAC_H_
+
+// This file defines the interface between the GC and the DAC. The interface consists of two things:
+//   1. A number of variables ("DAC vars") whose addresses are exposed to the DAC (see "struct GcDacVars")
+//   2. A number of types that are analogues to GC-internal types. These types expose a subset of the
+//      GC-internal type's fields, while still maintaining the same layout.
+// This interface is strictly versioned, see gcinterface.dacvars.def for more information.
+
+#define NUM_GC_DATA_POINTS             9
+#define MAX_COMPACT_REASONS_COUNT      11
+#define MAX_EXPAND_MECHANISMS_COUNT    6
+#define MAX_GC_MECHANISM_BITS_COUNT    2
+#define MAX_GLOBAL_GC_MECHANISMS_COUNT 6
+#define NUMBERGENERATIONS              4
+
+// TODO(segilles) - Implement this scheme for Server GC
+namespace SVR {
+    class heap_segment;
+    class gc_heap;
+}
+
+// Analogue for the GC heap_segment class, containing information regarding a single
+// heap segment.
+class dac_heap_segment {
+public:
+    uint8_t* allocated;
+    uint8_t* committed;
+    uint8_t* reserved;
+    uint8_t* used;
+    uint8_t* mem;
+    size_t flags;
+    dac_heap_segment* next;
+    uint8_t _padding1[8];
+    uint8_t* background_allocated;
+    uint8_t _padding2[40];
+};
+
+// Analogue for the GC generation class, containing information about the start segment
+// of a generation and its allocation context.
+class dac_generation {
+public:
+    gc_alloc_context allocation_context;
+    uint8_t _padding1[8];
+    dac_heap_segment* start_segment;
+    uint8_t _padding2[8];
+    uint8_t* allocation_start;
+    uint8_t _padding3[149];
```

Comment:
The sizes of these structures; and offsets of the interesting members in them should be in the DAC globals to avoid these magic padding values; (and/or these structures should be re-arranged such that all interesting fields for diagnostic are first or together).